app/controllers/admin/classification_featurings_controller.rb

```diff
@@ -3,10 +3,12 @@ class Admin::ClassificationFeaturingsController < Admin::BaseController
   before_action :load_featuring, only: %i[edit destroy]
 
   def index
-    filter_params = params.permit!.to_h.slice(:page, :type, :author, :organisation, :title).
-      merge(state: "published", classification: @classification.to_param)
-    @filter = Admin::EditionFilter.new(Edition, current_user, filter_params)
+    filter_params = params.permit(:type, :author, :organisation, :title, :page,
+                                  :topical_event_id, :topic_id, :state, :commit)
```

Comment:
I'm assuming these new parameters (topical_event_id, topic_id, etc) are added to prevent the warning about unpermitted parameters? 

The whole permit system doesn't really work too good at a root level and is much better on a nested structure (eg `params.require(:x).permit(:y)`)

I'd suggest rather than adding the extra things you might just want to define this manually with:

```
filter_params = { page: params[:page],
                  type: params[:type],
                  author: params[:author],
                  organisation: params[:organisation],
                  title: params[:title],
                  state: "published",
                  classification: @classification.to_param }
```

You might also find you can do this to avoid brakeman warnings:

```
params.slice(:page, :type, :author, :organisation, :title)
      .permit!
      .to_h
      .merge(state: "published", classification: @classification.to_param)
```

Where we slice before doing a blanket permit which means it's limited to a few options. I'm not sure if Brakeman is smart enough to catch this but it seemed to work on my quick test.

Reply:
Yeah looking in brakeman source code the `slice` then `permit!` looks good: https://github.com/presidentbeef/brakeman/blob/72b85bd0730699229c87feae2cd1d5d68b0f5f82/lib/brakeman/checks/check_mass_assignment.rb#L162

Reply:
>I'm assuming these new parameters (topical_event_id, topic_id, etc) are added to prevent the warning about unpermitted parameters?

Yup you're correct!

`slice` and `permit!` are now being used over permitting each param, nice suggestion!